src/task.ts

```diff
@@ -261,12 +263,17 @@ export abstract class CompoundTask extends DFTask {
      *  A sub-task of this task.
      */
     public handleCompletion(child: TaskBase): void {
-        if (!this.isPlayed) {
-            this.isPlayed = child.isPlayed;
-        }
+        this.trySetIsPlayed();
```

Comment:
From a design perspective, wouldn't it be easier to pass in the child here, same as `trySetValue()`? I understand for `WhenAll`, we would need to loop over all the children anyway, so we can ignore that argument, but for a `WhenAny`, we can just set the compound task's `isPlayed` to that of the child? Or am I missing something here?

Reply:
I think you're slightly onto something. For `WhenAny`, we can't just set the `isPlayed` flag to that of the child being passed in. _If_ we wanted to make us of a `child` argument, then we would need to guarantee that we only set `isPlayed` to be equal to the `isPlayed` value of the **first** child to complete. Alternatively, we could set `isPlayed` such that it may only go from `False` to `True` but not the other way around. If we were to use this, it would be an optimization for sure.

However, for now, I personally like the declarative style of this implementation because it gives me confidence in its correctness and I don't think we've had a performance complaint from this side. But yes, I think we could consider optimizing this scenario, although I'd rather make that part of a different PR. Would you like to file an issue for it?

Reply:
I'm not sure I understand this:

> If we wanted to make use of a child argument, then we would need to guarantee that we only set isPlayed to be equal to the isPlayed value of the first child to complete.

I thought we already made that guarantee? In order words, once _any_ child of a `WhenAny` completes, doesn't the value of the entire compound `WhenAny` task equal that _first_ completed child (including, I'm assuming, its `isPlayed` flag)? If that's not the case, then I'm not sure I understand the semantics of the `isPlayed` flag on a `WhenAny` task

Reply:
Either way, I don't want to be pedantic, so I won't block on this and I will trust that you know what you're doing here

Reply:
I've filed the request to optimize this logic here: https://github.com/Azure/azure-functions-durable-js/issues/425